pkg/services/provisioning/dashboards/file_reader_test.go

```diff
@@ -101,55 +100,97 @@ func TestDashboardFileReader(t *testing.T) {
 				reader, err := NewDashboardFileReader(cfg, logger)
 				So(err, ShouldBeNil)
 
+				bus.AddHandler("test", func(cmd *models.GetDashboardQuery) error {
+					return models.ErrDashboardNotFound
+				})
+
+				fakeService.
+					On(
+						"SaveFolderForProvisionedDashboards",
+						mock.MatchedBy(func(dash *dashboards.SaveDashboardDTO) bool {
+							return dash.Dashboard.Title == "Team A"
+						}),
+					).
+					Return(&models.Dashboard{Id: 1}, nil)
+
+				fakeService.
+					On("GetProvisionedDashboardData", "Default").
+					Return([]*models.DashboardProvisioning{}, nil)
+
+				fakeService.
+					On("SaveProvisionedDashboard", mock.Anything, mock.Anything).
+					Return(&models.Dashboard{}, nil).
+					Times(2)
+
 				err = reader.startWalkingDisk()
 				So(err, ShouldBeNil)
 
-				folders := 0
-				dashboards := 0
-
-				for _, i := range fakeService.inserted {
-					if i.Dashboard.IsFolder {
-						folders++
-					} else {
-						dashboards++
-					}
-				}
+				dash := &dashboards.SaveDashboardDTO{}
+				dash.Dashboard = models.NewDashboardFolder(cfg.Folder)
+				dash.Dashboard.IsFolder = true
+				dash.Overwrite = true
+				dash.OrgId = cfg.OrgId
 
-				So(folders, ShouldEqual, 1)
-				So(dashboards, ShouldEqual, 2)
+				fakeService.AssertExpectations(t)
 			})
 
+			// There was an issue with former test that it did not mock GetProvisionedDashboardData, which
+			// is the function that actually returns existing provisioned dashboards so it inserted a dashboard
+			// but not due to update but as a new dashboard
```

Comment:
Not sure if I can give a credit to the mocking lib with finding this, but as I had to figure out what calls were made, I noticed fairly quickly that some calls that should have been made were not. With the manual mock this was hidden because it worked the same way whether it was updating or inserting a new dashboard.